src/main/scala/com/microsoft/hyperspace/util/ResolverUtils.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.util
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.analysis.Resolver
+
+/**
+ * [[ResolverUtils]] provides utility functions to resolve strings based on spark's resolver.
+ */
+object ResolverUtils {
+
+  /**
+   * Return available string if required string can be resolved with it, based on spark resolver.
+   *
+   * @param resolver Resolver.
+   * @param requiredString The string that requires resolution.
+   * @param availableString Available string to resolve from.
+   * @return Optional available string if resolution is successful, else None
+   */
+  def resolve(
+      resolver: Resolver,
+      requiredString: String,
+      availableString: String): Option[String] = {
+    if (resolver(requiredString, availableString)) Some(availableString) else None
```

Comment:
I am just curious if this one-liner is consistent with our formatting rules. I thought we `if` always needs `{}`.

Reply:
thanks @pirz , I had my doubts with this one too. In [databricks scala guide](https://github.com/databricks/scala-style-guide#curly-braces), I saw that as long as everything fits in a line, it's acceptable. Our scala-style also did not complain for this so I kept it.